store/tikv/coprocessor.go

```diff
@@ -516,20 +519,23 @@ func (sender *copIteratorTaskSender) run() {
 	}
 }
 
-func (it *copIterator) recvFromRespCh(ctx context.Context, respCh <-chan *copResponse) (resp *copResponse, ok bool, exit bool) {
+func (it *copIterator) recvFromRespCh(ctx context.Context, respCh <-chan *copResponse) (resp *copResponse, ok, exit bool, err error) {
 	select {
 	case resp, ok = <-respCh:
 		if it.memTracker != nil && resp != nil {
 			it.memTracker.Consume(-int64(resp.MemSize()))
 		}
 	case <-it.finishCh:
 		exit = true
+		err = it.cancelErr
 	case <-ctx.Done():
 		// We select the ctx.Done() in the thread of `Next` instead of in the worker to avoid the cost of `WithCancel`.
 		if atomic.CompareAndSwapUint32(&it.closed, 0, 1) {
 			close(it.finishCh)
+			it.cancelErr = ctx.Err()
```

Comment:
Is the error only can be a cancel error (e.g: deadline exceed)? If not, the field name has may mislead maintainer.